lib/streamlit/__init__.py

```diff
@@ -44,6 +44,18 @@
 
 # IMPORTANT: Prefix with an underscore anything that the user shouldn't see.
 
+
+# Set Matplotlib backend to avoid a crash.
+# The default Matplotlib backend crashes Python on OSX when run on a thread
+# that's not the main thread, so here we set a safer backend as a fix.
+# This fix is OS-independent. We didn't see a good reason to make this
+# Mac-only. Consistency within Streamlit seemed more important.
+# This needs to run before any other import of matplotlib could happen.
+import os as _os
+
+_os.environ["MPLBACKEND"] = "Agg"
```

Comment:
nit: it'd probably be good to mention in the comment that this needs to stay where it is before any other imports to ensure that it occurs before matplotlib has a chance to be imported

Reply:
See the last line of comment above the os :) 

> This needs to run before any other import of matplotlib could happen.

I will copy this moment down above the `environ` and add an **Important**.